bitcoin/script/script.h

```diff
@@ -587,4 +573,27 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+
+    ([&] (Ts&& input) constexpr {
```

Comment:
I don't think this `constexpr` is useful. The body of the function always has the ability to perform allocations, so it can't ever actually be evaluatable at compile time.

Reply:
Interesting that it doesn't fail to compile